common/data_refinery_common/file_management.py

```diff
@@ -0,0 +1,178 @@
+
+
+def _get_downloaded_name(batch: Batch) -> str:
+    path = urllib.parse.urlparse(batch.download_url).path
+    return os.path.basename(path)
+
+
+def _get_processed_name(batch: Batch) -> str:
+    file_base = batch.name.split(".")[0]
+    return file_base + "." + batch.processed_format
+
+
+def get_raw_dir(batch: Batch) -> str:
+    if USE_S3:
+        return os.path.join(RAW_PREFIX, batch.internal_location)
+    else:
+        return os.path.join(LOCAL_ROOT_DIR, RAW_PREFIX, batch.internal_location)
+
+
+def get_raw_download_path(batch: Batch) -> str:
+    """Get the path to the downloaded file in the raw directory.
+
+    In cases where extraction is necessary, this will not match the
+    name of the batch's extracted file.
+    """
+    return os.path.join(get_raw_dir(batch), _get_downloaded_name(batch))
+
+
+def get_raw_path(batch: Batch) -> str:
+    return os.path.join(get_raw_dir(batch), batch.name)
+
+
+# The following four functions use the ID of the batch in the
+# temporary paths so it can be removed after processing is complete
+# without interfering with other jobs. An optional dir_name can be
+# used instead.
+def get_temp_dir(batch: Batch, dir_name: str = None) -> str:
+    dir_name = dir_name if dir_name is not None else str(batch.id)
+    return os.path.join(LOCAL_ROOT_DIR,
+                        TEMP_PREFIX,
+                        batch.internal_location,
+                        dir_name)
+
+
+def get_temp_download_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the downloaded file in the temp directory
+
+    In cases where extraction is necessary, this will not match the
+    name of the batch's extracted file.
+    """
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        _get_downloaded_name(batch))
+
+
+def get_temp_pre_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the pre-processed file for the batch."""
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        batch.name)
+
+
+def get_temp_post_path(batch: Batch, dir_name: str = None) -> str:
+    """Returns the path of the post-processed file for the batch."""
+    return os.path.join(get_temp_dir(batch, dir_name),
+                        _get_processed_name(batch))
+
+
+def get_processed_dir(batch: Batch) -> str:
+    if USE_S3:
+        return os.path.join(PROCESSED_PREFIX, batch.internal_location)
+    else:
+        return os.path.join(LOCAL_ROOT_DIR, PROCESSED_PREFIX, batch.internal_location)
+
+
+def get_processed_path(batch: Batch) -> str:
+    return os.path.join(get_processed_dir(batch),
+                        _get_processed_name(batch))
+
+
+def upload_raw_file(batch: Batch, dir_name: str = None) -> None:
+    """Moves the batch's raw file out of the temp directory.
+
+    Depending on the value of the USE_S3 environment variable this may
+    just be to the RAW_PREFIX directory or it may be to S3.
```

Comment:
This sentence is supposed to be modifying the one line description, but I can make it more clear by changing this to `Depending on the value of the USE_S3 environment variable this will either move the batch's raw file to the RAW_PREFIX directory or to S3.` if that is more clear. I don't want to use the word `upload` because if `USE_S3` is false then there won't be any uploading going on.